.github/workflows/save_cache.yml

```diff
@@ -0,0 +1,41 @@
+name: Create cache from commits on main
+
+on:
+  push:
+    branches:
+      - main
+      - chore-add-kind
+  workflow_dispatch:
+
+
+jobs:
+  save-poetry-cache:
+    runs-on: ubuntu-latest
+    env:
+      CACHE_KEY: main-branch-poetry-cache-ubuntu
+      CACHE_PATH: .devcontainer/.poetry_cache
+      DEVCONTAINER_IMAGE_CACHE: ghcr.io/swissdatasciencecenter/renku-data-services/devcontainer
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install python deps
+        uses: devcontainers/ci@v0.3
+        with:
+          runCmd: poetry install --with dev
+          push: always
+          skipContainerUserIdUpdate: false
+          imageName: ${{ env.DEVCONTAINER_IMAGE_CACHE }}
+          cacheFrom: ${{ env.DEVCONTAINER_IMAGE_CACHE }}
+      - uses: actions/cache/save@v3
+        name: Create cache
+        with:
+          path: ${{ env.CACHE_PATH }}
+          key: ${{ env.CACHE_KEY }}
```

Comment:
This is what I wanted. As in I dont want perfect caching but as long as we can reuse the cache from main but not change it that is fine. I am a bit scared to keep updating the cache all the time.

Also my understanding of the github caches is that they cannot be used across branches. So you can only use the cache from the current branch or from a parent branch. With this in mind having this more ideal caching seemed less important because in a fresh PR you can only use the cache from main. And I worried that if we add the hash of poetry.lock then we have even less chances of hits. But we can try this.

Reply:
the bullet point right below leafty's in the link she posted mentions that feature branches can share the cache of the main branch, just not from one feature branch in another.

Reply:
No, the point is that as is, you will create a single cache archive FOREVER. It will not update, which means that updates to the dependencies contained in poetry will not be cached after being merged to `main`.

Reply:
> I am a bit scared to keep updating the cache all the time.

This will not happen if the cache key reflects the `poetry.lock` contents.

Reply:
Ok I removed all the poetry/python caching. We can add that later.